### PR TITLE
#2240 Error handling of DICOM file in Detail

### PR DESCRIPTION
**Pull request by shirley9625**
Error handling of DICOM file in Detail(GUI), able to force open InvalidDicomError file; for NotImplementedError redirect to pydicom document

### CODE REVIEW

**Review by ryan-a-git — DISMISSED** [review]
Looks good, didn't see any testing though?

File: onko_dicom.py [hunk 1]
```diff
@@ -85,10 +106,56 @@ def open_file(self):
             logger.info("open_file user canceled open operation")
             return
 
-        self.plot_w.plot_dcm(full_path[0])
-        self.close_action.setEnabled(True)
-
-        logger.info("open_file completed within OnkoDicom")
+        """Error handling"""
```
Comment (pedrooste) [inline onko_dicom.py]: This has been added in from Liams right? we just need to be careful it doesn't clash :)

File: onko_dicom.py [hunk 2]
```diff
@@ -85,10 +106,56 @@ def open_file(self):
...
+                logger.error("(%s): ValueError. Error:(%s)", full_path[0], err)
+                response = self.err_msg('Error', 'NotImplementedError and ValueError. \n\nError: ' +
+                                        "Unable to decode pixel data  with a transfer syntax UID"
+                                        "as there are no pixel data handlers "
+                                        "available that support it. Please see the pydicom "
+                                        "documentation for information on supported transfer syntaxes\n\n"
+                                        + 'Error: ' + ''.join([str(err)])
+                                        )
+                if response == 'F':
+                    import webbrowser
```
Comment (pedrooste) [inline onko_dicom.py]: pylint, import from top level

File: plot_widget.py [hunk 3]
```diff
@@ -64,6 +64,21 @@ def plot_dcm(self, path):
         logger.info("plot_dcm completed within PlotWidget")
         return self.axes.axis() != (0.0, 1.0, 0.0, 1.0)
 
+    def force_plot_dcm(self, path):
```
Comment (pedrooste) [inline plot_widget.py]: reuse plot_dcm with the added attribute force. This can be set to a default of false

File: onko_dicom.py [hunk 4]
```diff
@@ -85,10 +106,56 @@ def open_file(self):
...
+            try:
+                logger.error("(%s): NotImplementedError. Error:(%s)", full_path[0], err)
+                self.err_msg('Error', 'NotImplementedError. \n\nError: ' + ''.join([str(err)]))
+            except ValueError as err:
+                logger.error("(%s): ValueError. Error:(%s)", full_path[0], err)
+                response = self.err_msg('Error', 'NotImplementedError and ValueError. \n\nError: ' +
+                                        "Unable to decode pixel data  with a transfer syntax UID"
+                                        "as there are no pixel data handlers "
+                                        "available that support it. Please see the pydicom "
+                                        "documentation for information on supported transfer syntaxes\n\n"
```
Comment (pedrooste) [inline onko_dicom.py]: syntax's

**Review by pedrooste — CHANGES_REQUESTED** [review]

File: onko_dicom.py (hunk 1 above)
Comment (shirley9625) [reply onko_dicom.py]: No, He didn't add

File: onko_dicom.py (hunk 4 above)
Comment (shirley9625) [reply onko_dicom.py]: lol, This is the original text of the pydicom error

File: onko_dicom.py (hunk 2 above)
Comment (shirley9625) [reply onko_dicom.py]: fixed

File: plot_widget.py (hunk 3 above)
Comment (shirley9625) [reply plot_widget.py]: I think this is easier to test, and not the same as the normal way to open. We can add some other stuff later and logging should also be distinguished from the normal one as this is force open.

Comment (pedrooste) [reply plot_widget.py]: In terms of testing, i think its actually easier, you would just need two tests for the same method. That is a good point about the logging, @ryan-a-git whatcha think?

File: onko_dicom.py (hunk 4 above)
Comment (pedrooste) [reply onko_dicom.py]: oh haha thats weird

File: plot_widget.py (hunk 3 above)
Comment (pedrooste) [reply plot_widget.py]: Talking to jack he reckons it should be the same function as well and pass force in. logging can be personalized by formatting the force parameter in

File: onko_dicom.py [hunk 5]
```diff
@@ -67,6 +73,24 @@ def create_menu(self):
...
+        button_y = msg_box.button(QMessageBox.Yes)
+        button_y.setText('Force Open')
+        button_n = msg_box.button(QMessageBox.No)
+        button_n.setText('Abort')
+        msg_box.exec()
+
+        if msg_box.clickedButton() == button_y:
+            return 'F'
+        elif msg_box.clickedButton() == button_n:
+            return 'A'
```
Comment (pedrooste) [inline onko_dicom.py]: I think this should return true/false, any reason your using F and A

Comment (shirley9625) [reply onko_dicom.py]: Not, these are only for error handling button response 
if response == ‘F’:
     return true

**Pull request closed without merging**